pkg/capture/GPPacket_test.go

```diff
@@ -12,7 +12,132 @@
+}
+
+type testCase struct {
+	input  testParams
+	epHash EPHash
+	isIPv4 bool
+}
+
+var testCases = []testParams{
+	{"2c04:4000::6ab", "2c01:2000::3", 0, 0, ICMPv6, [14]byte{0x80}, DirectionRemains},       // ICMPv6 echo request
+	{"2c01:2000::3", "2c04:4000::6ab", 0, 0, ICMPv6, [14]byte{0x81}, DirectionReverts},       // ICMPv6 echo reply
+	{"fe80::3df3:abbf:3d8d:7f03", "ff02::2", 0, 0, ICMPv6, [14]byte{0x86}, DirectionRemains}, // ICMPv6 router advertisement
+	{"10.0.0.1", "10.0.0.2", 0, 0, ICMP, [14]byte{0x08}, DirectionRemains},                   // ICMP echo request
+	{"10.0.0.2", "10.0.0.1", 0, 0, ICMP, [14]byte{0x00}, DirectionReverts},                   // ICMP echo reply
+	{"10.0.0.1", "10.0.0.2", 37485, 17500, TCP, [14]byte{}, DirectionRemains},                // TCP request to Dropbox LanSync from client on ephemeral port
+	{"10.0.0.2", "10.0.0.1", 17500, 34000, TCP, [14]byte{}, DirectionReverts},                // TCP response from Dropbox LanSync to client on ephemeral port
+	{"10.0.0.1", "4.5.6.7", 33561, 443, UDP, [14]byte{}, DirectionRemains},                   // UDP request from ephemaral port to privileged port
+	{"4.5.6.7", "10.0.0.1", 443, 33561, UDP, [14]byte{}, DirectionReverts},                   // UDP response from privileged port to ephemaral port
+	{"10.0.0.1", "4.5.6.7", 33561, 33560, UDP, [14]byte{}, DirectionRemains},                 // UDP request from higher ephemaral port to lower ephemaral port
+	{"4.5.6.7", "10.0.0.1", 33560, 33561, UDP, [14]byte{}, DirectionReverts},                 // UDP response from lower ephemaral port to higher ephemaral port
+	{"10.0.0.1", "4.5.6.7", 444, 443, UDP, [14]byte{}, DirectionRemains},                     // UDP request from higher privileged port to lower privileged port
+	{"4.5.6.7", "10.0.0.1", 443, 444, UDP, [14]byte{}, DirectionReverts},                     // UDP response from lower privileged port to higher privileged port
+	{"10.0.0.1", "4.5.6.7", 33561, 33561, UDP, [14]byte{}, DirectionRemains},                 // UDP packet from identical ephemaral ports (no change, have to assume this is the first packet)
+	{"10.0.0.1", "4.5.6.7", 443, 443, UDP, [14]byte{}, DirectionRemains},                     // UDP packet from identical privileged ports (no change, have to assume this is the first packet)
+	{"10.0.0.1", "4.5.6.7", 0, 53, UDP, [14]byte{}, DirectionRemains},                        // DNS request from NULLed ephemaral port
+	{"2c04:4000::6ab", "2c04:4000::6ab", 33561, 33561, UDP, [14]byte{}, DirectionRemains},    // UDP packet from identical ephemaral ports (no change, have to assume this is the first packet)
+	{"2c04:4000::6ab", "2c04:4000::6ab", 443, 443, UDP, [14]byte{}, DirectionRemains},        // UDP packet from identical privileged ports (no change, have to assume this is the first packet)
+	{"2c04:4000::6ab", "2c04:4000::6ab", 0, 53, UDP, [14]byte{}, DirectionRemains},           // DNS request from NULLed ephemaral port
```

Comment:
I’d add DHCP port 67 here as a test case due to the fact that the DHCP server listens on port 67 and clients send from 68.